compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -160,6 +161,15 @@ private boolean getSticky(PackageContext rootPackageContext) {
             this.autoUpdate = false;
             return true;
         }
+
+        // if distribution is not same, we anyway return sticky as false
+        String distVersion = rootPackageContext.packageManifest().ballerinaVersion();
```

Comment:
This will end up discarding the build file for every run when the dist version in Ballerina.toml is an old one. I think we need to persist the dist version to the `target/build` file. Not having the dist version should only be a one-time thing. We should persist the dist version to the build file when updating it.

Reply:
@hevayo wdyt??

Reply:
Yes it should be in build file

Reply:
Added